File: Core/Entities/Movie.cs
namespace Core.Entities;

public class Movie : BaseEntity
{
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public string? Director { get; set; }
    public string? PictureUrl { get; set; }
    public int Rating { get; set; }
    public int ReleaseYear { get; set; }

    // Navigation properties
    public IEnumerable<MovieCategory> MovieCategories { get; set; } = null!;
    public IEnumerable<UserReview> UserReviews { get; set; } = null!;
}

File: clientApp/src/app/shared/models/movie.ts
import { ICategory } from "./category";
import { IUserReview } from "./userReview";

export interface IMovie {
    id: number;
    title: string;
    description: string;
    director: string;
    pictureUrl: string;
    rating: number;
    releaseYear: number;
    categories: ICategory[];
    userReviews: IUserReview[];
}